atd-vze/src/views/Tools/CreateCrashRecord.js

```diff
@@ -14,22 +14,70 @@ import {
   Form,
   FormGroup,
   FormText,
+  FormFeedback,
 } from "reactstrap";
 import moment from "moment";
 import { withApollo } from "react-apollo";
 import { gql } from "apollo-boost";
 import "./CreateCrashRecord.css";
+import UnitsForm from "./UnitsForm";
 
 const CreateCrashRecord = ({ client }) => {
-  const [tempId, setTempId] = useState(1000);
-  const [caseId, setCaseId] = useState("");
-  const [fatalityCount, setFatalityCount] = useState(0);
-  const [susSeriousInjuryCount, setSusSeriousInjuryCount] = useState(0);
+  const unitsInitialState = [
+    { unit_desc_id: 1, atd_fatality_count: 0, sus_serious_injry_cnt: 0 },
+  ];
+
+  const formInitialState = {
+    tempId: 1000,
+    caseId: "",
+    crashTime: moment(new Date()).format("HH:mm:ss"),
+    crashDate: moment(new Date()).format("YYYY-MM-DD"),
+    primaryAddress: "",
+    secondayAddress: "",
+    units: [
+      { unit_desc_id: 1, atd_fatality_count: 0, sus_serious_injry_cnt: 0 },
+    ],
+  };
+
+  const [tempId, setTempId] = useState(formInitialState.tempId);
+  const [caseId, setCaseId] = useState(formInitialState.caseId);
   const [successfulNewRecordId, setSuccessfulNewRecordId] = useState(null);
-  const [crashDate, setCrashDate] = useState(
-    moment(new Date()).format("YYYY-MM-DD")
-  );
+  const [crashTime, setCrashTime] = useState(formInitialState.crashTime);
+  const [crashDate, setCrashDate] = useState(formInitialState.crashDate);
   const [feedback, setFeedback] = useState(false);
+  const [primaryAddress, setPrimaryAddress] = useState(
+    formInitialState.primaryAddress
+  );
+  const [secondayAddress, setSecondaryAddress] = useState(
+    formInitialState.secondayAddress
+  );
+
+  function unitsReducer(state, action) {
+    const { type, payload, unitIndex } = action;
+
+    // When a user clicks the "Add Unit" button, add another unit object to the
+    // unit state array using the initial state template provided from the payload.
+    if (type === "addNewUnit") {
+      return [...state, payload];
+    }
+
+    if (type === "reset") {
+      return unitsInitialState;
+    }
+
+    const updatedState = [...state];
+    updatedState[unitIndex][type] = payload;
+    return updatedState;
+  }
+
+  const [unitFormState, unitFormDispatch] = useReducer(
+    unitsReducer,
+    unitsInitialState
+  );
+
+  function isFieldInvalid(data) {
+    return !data;
+  }
 
   useEffect(() => {
```

Comment:
I didn't catch this the first time around, but, since this `useEffect` doesn't have a dependency array, it is querying for the highest temp record each time a form state is updated. If that isn't intentional, it could be limited to one request by passing `client` or an empty array (which causes a linter warning).

Reply:
nice, thanks for this tip. I didn't realize that!